programs/bubblegum/program/src/lib.rs

```diff
@@ -1348,6 +1597,114 @@ pub mod bubblegum {
         )
     }
 
+    pub fn update_metadata<'info>(
+        ctx: Context<'_, '_, '_, 'info, UpdateMetadata<'info>>,
+        root: [u8; 32],
+        nonce: u64,
+        index: u32,
+        old_metadata: Option<MetadataArgs>,
+        new_name: Option<String>,
+        new_symbol: Option<String>,
+        new_uri: Option<String>,
+        new_creators: Option<Vec<Creator>>,
+        new_seller_fee_basis_points: Option<u16>,
+        new_primary_sale_happened: Option<bool>,
+        new_is_mutable: Option<bool>,
+    ) -> Result<()> {
+        assert_signed_by_tree_delegate(&ctx.accounts.tree_authority, &ctx.accounts.tree_delegate)?;
+
+        // Determine how the user opted to pass in the old MetadataArgs
+        let old_metadata = fetch_old_metadata_args(old_metadata, &ctx.accounts.old_metadata_acct)?;
+
+        // NFTs which are linked to verified collections cannot be updated through this instruction
+        require!(
+            old_metadata.collection.is_none()
+                || !old_metadata.collection.as_ref().unwrap().verified,
+            BubblegumError::NFTLinkedToCollection
+        );
+
+        process_update_metadata(
+            &ctx.accounts.merkle_tree.to_account_info(),
+            &ctx.accounts.leaf_owner,
+            &ctx.accounts.leaf_delegate,
+            &ctx.accounts.compression_program.to_account_info(),
+            &ctx.accounts.tree_authority.to_account_info(),
+            *ctx.bumps.get("tree_authority").unwrap(),
+            &ctx.accounts.log_wrapper,
+            ctx.remaining_accounts,
+            root,
+            old_metadata,
+            new_name,
+            new_symbol,
+            new_uri,
+            new_creators,
+            new_seller_fee_basis_points,
+            new_primary_sale_happened,
+            new_is_mutable,
+            nonce,
+            index,
+        )
+    }
+
+    pub fn update_metadata_collection_nft<'info>(
```

Comment:
This instruction might not be needed.

Reply:
Will remove from our branch as follow-on PR.